### ISSUE TITLE
Denoised BOLD abnormalities when last frame (maybe also first frame?) of run is FD outlier

### ISSUE THREAD

**Issue by madisoth** [PennLINC/xcp_d#973]
## Summary

When the last frame of a run is an FD outlier, denoised BOLD timeseries derivatives (both "desc-denoised*_bold" and "desc-interpolated_bold") have massively increasing spatial standard deviation and DVARS near the end of the run. This does not occur when FD censoring is disabled. Might be an issue with how outlier frames at start/end of run are replaced?  

## Additional details

- xcp_d version: 0.5.0rc2; also seen in 0.4.1rc3.dev16+gb52fcc3
- Docker version:
- Singularity version: Apptainer 1.2.2-1.el7

### What were you trying to do?

Process a subject with this run command:

```
/usr/local/miniconda/bin/xcp_d /fmriprep_out /xcpd_out participant -f 0.3 --cifti --participant-label XXXXXX
--lower-bpf 0.009 --upper-bpf 0.25 -p 36P --smoothing 2 --band-stop-min 11 --band-stop-max 17 --motion-filter-type
notch --omp-nthreads 3 --nthreads 24 --despike --warp-surfaces-native2std -w /wkdir
```

### What actually happened?
 
In the executive summary, DVARS increased greatly near the end of the runs where the last frame was an FD outlier, and in the executive summary the parcellated matrices look "blown out" (the values are abnormally large, either positive or negative).

## Reproducing the bug

Reprocessing the same subject(s) with FD censoring disabled, which did not show the same issues

**Comment by tsalo** [comment]
@madisoth yes this is reported in #949 and fixed in #950. Can you try 0.5.2?

**Comment by madisoth** [comment]
@tsalo  Looks all good in 0.5.2 -- thanks!
